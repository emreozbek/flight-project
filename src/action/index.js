import { ADD_FLIGHT, FETCH_FLIGHTS, SHOW_ERROR } from "../constant";

export const addFlight = data => ({
  type: ADD_FLIGHT,
  data
});
export const fetchFlights = () => ({
  type: FETCH_FLIGHTS
});
export const showError = data => ({
  type: SHOW_ERROR,
  data
});


