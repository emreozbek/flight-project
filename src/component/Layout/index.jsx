import React, { Fragment } from "react";
import PropTypes from "prop-types";
import { Paper, Typography } from "@material-ui/core";
import { withStyles } from "@material-ui/core/styles";
import { styles } from "./styles";

export function Layout(props) {
  const { background, wrapper } = props.classes;
  return (
    <Fragment>
      <div className={background} />
      <div className={wrapper}>
        <Typography variant="h5" component="h1" gutterBottom>
          Flight Project
        </Typography>
        <Paper elevation={1}>{props.children}</Paper>
        <Typography variant="caption">
          Powered using React, Redux, Redux-Saga and Axios
        </Typography>
      </div>
    </Fragment>
  );
}
Layout.propTypes = {
  classes: PropTypes.object.isRequired
};
export default withStyles(styles)(Layout);
