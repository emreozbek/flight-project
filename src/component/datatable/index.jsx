import React, { Component, Fragment } from "react";
import PropTypes from "prop-types";
import _ from "lodash";
import {
  Table,
  TableBody,
  TableCell,
  TablePagination,
  TableRow,
  withStyles,
  InputBase,
  InputAdornment
} from "@material-ui/core";
import SearchIcon from "@material-ui/icons/Search";

import DataTableHead from "./Head";
import { styles } from "./styles";
import { getLocalStorage, setLocalStorage } from "../../utility/storage";

export class DataTable extends Component {
  defaults = {
    order: "asc",
    orderBy: this.props.fields[0].id,
    page: 0,
    rowsPerPage: 5,
    rowsPerPageOptions: [5, 10, 25, 50],
    search: ""
  };
  state = _.assign(
    {},
    this.defaults,
    this.props.options,
    JSON.parse(getLocalStorage(this.props.id))
  );
  setStorage() {
    setLocalStorage(this.props.id, JSON.stringify(this.state));
  }
  handleChangePage = (event, page) => {
    this.setState({ page }, this.setStorage);
  };
  handleChangeRowsPerPage = event => {
    this.setState({ rowsPerPage: event.target.value }, this.setStorage);
  };
  handleRequestSort = (event, orderBy) => {
    let order = "desc";
    if (this.state.orderBy === orderBy && this.state.order === "desc") {
      order = "asc";
    }
    this.setState({ order, orderBy }, this.setStorage);
  };
  filter(data, text) {
    return _.filter(data, row => {
      let isContain = false;
      _.forEach(row, value => {
        if (_.includes(_.toUpper(value), _.toUpper(text))) {
          isContain = true;
          return false;
        }
      });
      return isContain;
    });
  }
  render() {
    const { fields, classes } = this.props;
    const {
      order,
      orderBy,
      rowsPerPage,
      page,
      rowsPerPageOptions,
      search
    } = this.state;
    const data = this.filter(this.props.data, search);

    return (
      <Fragment>
        <div className={classes.searchBox}>
          <InputBase
            className={classes.input}
            placeholder="Search in the list"
            defaultValue={search}
            onChange={event =>
              this.setState({ search: event.target.value }, this.setStorage)
            }
            endAdornment={
              <InputAdornment position="end">
                <SearchIcon />
              </InputAdornment>
            }
            fullWidth
          />
        </div>

        <div className={classes.tableWrapper}>
          <Table aria-labelledby="tableTitle" className={classes.table}>
            <DataTableHead
              fields={fields}
              order={order}
              orderBy={orderBy}
              onRequestSort={this.handleRequestSort}
            />
            <TableBody>
              {_.map(
                _.slice(
                  _.orderBy(data, [orderBy], [order]),
                  page * rowsPerPage,
                  page * rowsPerPage + rowsPerPage
                ),
                n => {
                  return (
                    <TableRow tabIndex={-1} key={n.id}>
                      {_.flatMap(fields, (field, index) => (
                        <TableCell key={index}>{n[field.id]}</TableCell>
                      ))}
                    </TableRow>
                  );
                }
              )}
            </TableBody>
          </Table>
        </div>
        {data.length > 0 && (
          <TablePagination
            rowsPerPageOptions={rowsPerPageOptions}
            component="div"
            count={data.length}
            rowsPerPage={rowsPerPage}
            page={page}
            backIconButtonProps={{
              "aria-label": "Previous Page"
            }}
            nextIconButtonProps={{
              "aria-label": "Next Page"
            }}
            onChangePage={this.handleChangePage}
            onChangeRowsPerPage={this.handleChangeRowsPerPage}
          />
        )}
      </Fragment>
    );
  }
}

DataTable.propTypes = {
  id: PropTypes.string.isRequired,
  fields: PropTypes.array.isRequired,
  data: PropTypes.array.isRequired,
  options: PropTypes.object,
  classes: PropTypes.object
};

export default withStyles(styles)(DataTable);
