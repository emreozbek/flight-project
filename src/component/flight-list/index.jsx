import React, { Fragment } from "react";
import { connect } from "react-redux";
import DataTable from "../datatable";
import Loading from "../loading";

export function List(props) {
  return (
    <Fragment>
      <Loading />
      <DataTable
        id="flightList"
        fields={[
          { id: "flightType", label: "Flight Type" },
          { id: "departure", label: "Departure" },
          { id: "departureDate", label: "Departure Date" },
          { id: "arrival", label: "Arrival" },
          { id: "arrivalDate", label: "Arrival Date" }
        ]}
        data={props.data}
      />
    </Fragment>
  );
}
const mapStateToProps = state => ({ data: state.reducer.data });
export default connect(mapStateToProps)(List);
