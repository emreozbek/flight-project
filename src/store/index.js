import { applyMiddleware, combineReducers, createStore } from "redux";
import { routerMiddleware, routerReducer } from "react-router-redux";
import { reducer as formReducer } from "redux-form";
import createSagaMiddleware from "redux-saga";
import { createBrowserHistory } from "history";
import reducer from "../reducer";
import Saga from "../saga";

const sagaMiddleware = createSagaMiddleware();
export const history = createBrowserHistory();
export const store = createStore(
  combineReducers({
    reducer,
    routing: routerReducer,
    form: formReducer
  }),
  applyMiddleware(sagaMiddleware, routerMiddleware(history))
);
sagaMiddleware.run(Saga);
