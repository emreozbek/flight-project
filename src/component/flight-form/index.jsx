import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { Field, reduxForm } from "redux-form";
import { Grid, Button, withStyles } from "@material-ui/core";
import moment from "moment";
import { addFlight } from "../../action";
import {
  renderTextField,
  renderRadioGroup,
  renderDateTimePicker
} from "../../utility/field-renderer";
import { styles } from "./styles";
import { DATE_FORMAT, MESSAGE } from "../../config";
import { validate } from "./validate";

export class FlightForm extends Component {
  render() {
    const { classes, addFlight, reset, handleSubmit } = this.props;
    return (
      <form
        onSubmit={handleSubmit(values => {
          values.departureDate = moment(values.departureDate).format(
            DATE_FORMAT
          );
          values.arrivalDate = moment(values.arrivalDate).format(DATE_FORMAT);
          addFlight(values);
          reset();
        })}
        className={classes.padding}
        noValidate="novalidate"
      >
        <Grid container spacing={16}>
          <Grid item xs={12} sm={6} className={classes.padding}>
            <Field
              required
              helperText={MESSAGE.form.required}
              name="departure"
              component={renderTextField}
              label="Departure"
              margin="dense"
              fullWidth
            />
          </Grid>
          <Grid item xs={12} sm={6} className={classes.padding}>
            <Field
              required
              name="arrival"
              helperText={MESSAGE.form.required}
              component={renderTextField}
              label="Arrival"
              margin="dense"
              fullWidth
            />
          </Grid>
          <Grid item xs={12} sm={6} className={classes.padding}>
            <Field
              name="departureDate"
              component={renderDateTimePicker}
              helperText={MESSAGE.form.required}
              label="Departure Date"
              margin="dense"
              fullWidth
              required
            />
          </Grid>
          <Grid item xs={12} sm={6} className={classes.padding}>
            <Field
              name="arrivalDate"
              component={renderDateTimePicker}
              helperText={MESSAGE.form.required}
              label="Arrival Date"
              margin="dense"
              fullWidth
              required
            />
          </Grid>
          <Grid item xs={12} sm={12} className={classes.padding}>
            <Field
              name="flightType"
              label="Flight Type"
              component={renderRadioGroup}
              options={[
                { value: "Cheap", label: "Cheap" },
                { value: "Business", label: "Business" }
              ]}
              row
            />
          </Grid>
        </Grid>
        <Button variant="outlined" color="primary" type="submit">
          Save
        </Button>
        <Button
          color="secondary"
          className={classes.marginLeft}
          onClick={() => {
            reset();
          }}
        >
          Reset
        </Button>
      </form>
    );
  }
}
FlightForm.propTypes = {
  classes: PropTypes.object,
  addFlight: PropTypes.func,
  handleSubmit: PropTypes.func
};
const flightForm = reduxForm({
  form: "flight",
  destroyOnUnmount: false,
  validate
})(FlightForm);
const mapDispatchToProps = {
  addFlight
};
export default connect(
  null,
  mapDispatchToProps
)(withStyles(styles)(flightForm));
