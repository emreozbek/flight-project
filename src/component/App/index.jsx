import React, { Component } from "react";
import { Route, Switch } from "react-router-dom";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import Theme from "../../asset/theme";
import Layout from "../Layout";
import TabMenu from "../tab-menu";
import NotFound from "../not-found";
import { fetchFlights } from "../../action";
import Snackbar from "../snackbar";

export class App extends Component {
  componentDidMount() {
    this.props.fetchFlights();
  }
  render = () => (
    <Theme>
      <Layout>
        <Snackbar />
        <Switch>
          <Route path="/" exact component={TabMenu} />
          <Route path="/add" component={TabMenu} />
          <Route component={NotFound} />
        </Switch>
      </Layout>
    </Theme>
  );
}
App.propTypes = {
  fetchFlights: PropTypes.func
};

const mapStateToProps = state => ({ ...state });
const mapDispatchToProps = {
  fetchFlights
};
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(App);
