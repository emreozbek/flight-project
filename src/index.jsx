import React from "react";
import { render } from "react-dom";
import { Provider } from "react-redux";
import { Router } from "react-router";
import App from "./component/App";
import { store, history } from "./store";

render(
  <Provider store={store}>
    <Router history={history}>
      <App />
    </Router>
  </Provider>,
  document.getElementById("app")
);
