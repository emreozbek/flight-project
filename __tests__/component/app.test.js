import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { store } from "../../src/store";
import { App } from "../../src/component/App";
import { Router } from "react-router";
import { history } from "../../src/store";

describe("App", () => {
  it("renders correctly", () => {
    const mockFunction = jest.fn();
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router history={history}>
            <App fetchFlights={mockFunction} />
          </Router>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
