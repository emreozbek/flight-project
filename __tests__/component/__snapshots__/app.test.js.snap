// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
Array [
  <div
    className="Layout-background-1"
  />,
  <div
    className="Layout-wrapper-2"
  >
    <h1
      className="MuiTypography-root-3 MuiTypography-h5-19 MuiTypography-gutterBottom-30"
    >
      Flight Project
    </h1>
    <div
      className="MuiPaper-root-39 MuiPaper-elevation1-42 MuiPaper-rounded-40"
    >
      <div
        className="MuiTabs-root-76"
      >
        <div
          className="MuiTabs-flexContainer-77"
        >
          <div
            className="MuiTabs-scroller-79 MuiTabs-fixed-80"
            onScroll={[Function]}
            role="tablist"
            style={
              Object {
                "marginBottom": 0,
              }
            }
          >
            <div
              className="MuiTabs-flexContainer-77 MuiTabs-centered-78"
            >
              <a
                aria-selected={true}
                className="MuiButtonBase-root-97 MuiTab-root-85 MuiTab-textColorInherit-87 MuiTab-selected-90 MuiTab-fullWidth-92"
                href="/"
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="tab"
                tabIndex="0"
              >
                <span
                  className="MuiTab-wrapper-93"
                >
                  <span
                    className="MuiTab-labelContainer-94"
                  >
                    <span
                      className="MuiTab-label-95"
                    >
                      Flights
                    </span>
                  </span>
                </span>
                <span
                  className="MuiTouchRipple-root-162"
                />
              </a>
              <a
                aria-selected={false}
                className="MuiButtonBase-root-97 MuiTab-root-85 MuiTab-textColorInherit-87 MuiTab-fullWidth-92"
                href="/add"
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="tab"
                tabIndex="0"
              >
                <span
                  className="MuiTab-wrapper-93"
                >
                  <span
                    className="MuiTab-labelContainer-94"
                  >
                    <span
                      className="MuiTab-label-95"
                    >
                      Add FLight
                    </span>
                  </span>
                </span>
                <span
                  className="MuiTouchRipple-root-162"
                />
              </a>
            </div>
            <span
              className="MuiPrivateTabIndicator-root-100 MuiPrivateTabIndicator-colorSecondary-102 MuiTabs-indicator-84"
              style={
                Object {
                  "left": 0,
                  "width": 0,
                }
              }
            />
          </div>
        </div>
      </div>
      <div
        className="TabMenu-content-75"
      >
        <div
          className="overlay"
        >
          <div
            className="loading-animation"
          >
            <div
              className="lds-grid"
            >
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
            </div>
          </div>
        </div>
        <div
          className="DataTable-searchBox-105"
        >
          <div
            className="MuiInputBase-root-106 MuiInputBase-fullWidth-115 MuiInputBase-adornedEnd-111"
            onClick={[Function]}
          >
            <input
              className="MuiInputBase-input-116 MuiInputBase-inputAdornedEnd-122"
              defaultValue=""
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Search in the list"
              type="text"
            />
            <div
              className="MuiInputAdornment-root-123 MuiInputAdornment-positionEnd-126"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-128"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          className="DataTable-tableWrapper-104"
        >
          <table
            aria-labelledby="tableTitle"
            className="MuiTable-root-137 DataTable-table-103"
          >
            <thead
              className="MuiTableHead-root-138"
            >
              <tr
                className="MuiTableRow-root-139 MuiTableRow-head-142"
              >
                <th
                  aria-sort="ascending"
                  className="MuiTableCell-root-144 MuiTableCell-head-145"
                  scope="col"
                >
                  <span
                    className="MuiButtonBase-root-97 MuiTableSortLabel-root-156 MuiTableSortLabel-active-157"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onContextMenu={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    role="button"
                    tabIndex="0"
                  >
                    Flight Type
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-128 MuiTableSortLabel-icon-158 MuiTableSortLabel-iconDirectionAsc-160"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                      />
                    </svg>
                  </span>
                </th>
                <th
                  aria-sort={null}
                  className="MuiTableCell-root-144 MuiTableCell-head-145"
                  scope="col"
                >
                  <span
                    className="MuiButtonBase-root-97 MuiTableSortLabel-root-156"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onContextMenu={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    role="button"
                    tabIndex="0"
                  >
                    Departure
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-128 MuiTableSortLabel-icon-158 MuiTableSortLabel-iconDirectionAsc-160"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                      />
                    </svg>
                  </span>
                </th>
                <th
                  aria-sort={null}
                  className="MuiTableCell-root-144 MuiTableCell-head-145"
                  scope="col"
                >
                  <span
                    className="MuiButtonBase-root-97 MuiTableSortLabel-root-156"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onContextMenu={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    role="button"
                    tabIndex="0"
                  >
                    Departure Date
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-128 MuiTableSortLabel-icon-158 MuiTableSortLabel-iconDirectionAsc-160"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                      />
                    </svg>
                  </span>
                </th>
                <th
                  aria-sort={null}
                  className="MuiTableCell-root-144 MuiTableCell-head-145"
                  scope="col"
                >
                  <span
                    className="MuiButtonBase-root-97 MuiTableSortLabel-root-156"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onContextMenu={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    role="button"
                    tabIndex="0"
                  >
                    Arrival
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-128 MuiTableSortLabel-icon-158 MuiTableSortLabel-iconDirectionAsc-160"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                      />
                    </svg>
                  </span>
                </th>
                <th
                  aria-sort={null}
                  className="MuiTableCell-root-144 MuiTableCell-head-145"
                  scope="col"
                >
                  <span
                    className="MuiButtonBase-root-97 MuiTableSortLabel-root-156"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onContextMenu={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    role="button"
                    tabIndex="0"
                  >
                    Arrival Date
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-128 MuiTableSortLabel-icon-158 MuiTableSortLabel-iconDirectionAsc-160"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                      />
                    </svg>
                  </span>
                </th>
              </tr>
            </thead>
            <tbody
              className="MuiTableBody-root-161"
            />
          </table>
        </div>
      </div>
    </div>
    <span
      className="MuiTypography-root-3 MuiTypography-caption-13"
    >
      Powered using React, Redux, Redux-Saga and Axios
    </span>
  </div>,
]
`;
